core/http/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/transaction/Transaction.java

```diff
@@ -113,13 +117,12 @@ UUID getID() {
 	void begin(IsolationLevel level)
 		throws InterruptedException, ExecutionException
 	{
-		Future<Boolean> result;
-		synchronized (futures) {
-			result = executor.submit(() -> {
-				txnConnection.begin(level);
-				return true;
-			});
+		Future<Boolean> result = executor.submit(() -> {
+			txnConnection.begin(level);
+			return true;
+		});
 
+		synchronized (futures) {
```

Comment:
I will add some locking to ensure that there is concurrent access from the thread executor. Putting the .execute instruction inside of synchronized does nothing for thread-safety purposes, so I will add a semaphore to the executes to ensure they are complete before the next one starts.